changesLog.js

```diff
@@ -1,34 +1,75 @@
+const octokit = require('./githubAPI/octokit');
 const commander = require('commander');
-const { getIssuesByMilestone } = require('./changesLogGenerator');
+const fs = require('fs');
+const { fetchIssuesInMilestone, generateChangesLog } = require('./helpers');
 
-function commaSeparatedList(value, dummyPrevious) {
+function commaSeparatedList(value) {
   return value.split(',');
 }
 
+function checkValidStates(value) {
+  return (value === 'closed') || (value === 'open') || (value === 'all') ? value : undefined;
+}
+
 commander
   .version('1.0.0', '-v, --version')
   .usage('[OPTIONS]...')
-  .requiredOption('-m, --milestone <name>', 'must filter issue from milestone')
-  .option('-d, --duplicate <allow>', 'Should allow duplicated issues (Default: true)')
-  .option('-f, --filters <items>', 'comma separeted labels to filter issues', commaSeparatedList)
-  .option('-g, --groups <items>', 'comma separated labels to group by issues', commaSeparatedList)
+  .requiredOption('-m, --milestone <number>', 'must filter issue from milestone')
+  .option('-c, --customer <customer>, will filter the issues related to a customer if this option is received')
+  .option('-i, --includeAll <flag>', 'will include a section with issues done for all clients (Default: false)')
+  .option('-s, --state <state>', 'should use issues with state: closed, open, all (Default: closed)', checkValidStates)
+  .option('-f, --filters <labels>', 'comma separated labels to filter and group by issues', commaSeparatedList)
+  .option('-d, --duplicate <allow>', 'should allow duplicated issues in different groups (Default: true)')
   .parse(process.argv);
 
+let groupedIssues = [];
+const filters = commander.filters;
 const params = {
-  duplicate: commander.duplicate ? false : true,
   milestone: commander.milestone,
-  filters: commander.filters,
-  groups: commander.groups
+  state: commander.state ? commander.state : 'closed',
+  customer: commander.customer ? commander.customer : undefined,
+  duplicateIssues: commander.duplicate ? commander.duplicate : true,
+  includeIssueForAll: commander.includeAll ? commander.includeAll : true
 };
 
-console.log(`Generating Changelog for Milestone ${params.milestone}...`)
-console.log(`Duplicate issues (in different groups) set to: ${params.duplicate}.`)
-if (params.filters !== undefined) console.log(`Filters: ${params.filters}`);
-if (params.groups !== undefined) console.log(`Groups: ${params.groups}`);
+async function fetchIssues() {
+  console.log(`\n\nFetching issues in Milestone ${params.milestone}...\n`);
+  if(filters) {
+    groupedIssues = await fetchIssuesInMilestone(octokit, params, filters);
+    
+    if (groupedIssues.length > 0) {
+      console.log(`\nFinished fetching issues by filters in Milestone ${params.milestone}!\n`);
+      console.log(`Duplicate issues (in different groups) set to: ${params.duplicateIssues}.\n`);
```

Comment:
I think can remove the `'in different groups'` explanation? IMO good to keep the logs as concise as possible, and my thinking is that we should expect new users to read up on the `duplicateIssues` flag (using `-h` or docs) before running the changelog?

Reply:
Cool. I will change that